### PR TITLE
DOC: Improved doc for `Index.equals`

### LINKED ISSUE #33285: DOC: Mention Index.equals takes into account order of arguments
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandas.
- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

#### Code Sample, a copy-pastable example

```python
import pandas as pd
pd.Index(['10607', '6574', '6621', '99917']).equals(pd.Index(['99917', '10607', '6574', '6621'])) # False
set(['10607', '6574', '6621', '99917']) == set(['99917', '10607', '6574', '6621']) # True
```

#### Problem description

I have seen #13708, but according to documentation:

> Determine if two Index objects contain the same elements.

Should it return `True`  instead?

Reading how the function is implemented:

https://github.com/pandas-dev/pandas/blob/3adf3340453d6704d4a2cb47058214cc697a7d29/pandas/core/indexes/base.py#L4072-L4074

It relies on:

https://github.com/pandas-dev/pandas/blob/3adf3340453d6704d4a2cb47058214cc697a7d29/pandas/core/dtypes/missing.py#L421-L431

Which is indeed strict and do take order into account.

#### Expected Output

As I understand the equality of index is about having the same elements. It is not required the elements having the same order. I have the feeling that once pandas worked this way (maybe before `v0.23`).

What is the intended way ? Maybe the documentation should be a bit more explicit about this comparison. If you indicate me where I can update the doc, I will be glad to edit it.

#### Output of ``pd.show_versions()``

<details>

commit           : None
python           : 3.6.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-91-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.3
numpy            : 1.18.2
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 9.0.1
setuptools       : 46.1.3
Cython           : 0.29.14
pytest           : 5.3.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.1.8
lxml.etree       : 4.3.4
html5lib         : 0.999999999
pymysql          : None
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.11.1
IPython          : 7.13.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.4
matplotlib       : 3.2.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.13.0
pytables         : None
pytest           : 5.3.2
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.15
tables           : None
tabulate         : 0.8.3
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.8
numba            : None

</details>

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] closes #33285
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
...
+
+        Examples
+        --------
+        >>> idx1 = pd.Index([1, 2, 3])
+        >>> idx1
+        Int64Index([1, 2, 3], dtype='int64')
+        >>> idx1.equals(pd.Index([1, 2, 3]))
+        True
+
+        The dtype is been compared as well
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4214]: `is been` -> `is`

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
...
+
+        >>> idx2 = pd.Index(["1", "2", "3"])
+        >>> idx2
+        Index(['1', '2', '3'], dtype='object')
+
+        >>> idx1.equals(idx2)
+        False
+
+        The oreder is also been compared
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:4223]: `oreder is also been compared` -> `order is also compared`

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
             # coerces to object
             return self.astype(object).putmask(mask, value)
 
-    def equals(self, other) -> bool:
+    def equals(self, other: Any) -> bool:
         """
-        Determine if two Index objects contain the same elements.
+        Determine if two Index objects contain the same elements with the same order.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py:4198]: Or we could also simplify "... contain the same elements with the same order" to "... are equal" ?

We can in a second paragraph detail on what is considered as equal

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py:4214]: The dtype is actually *not* compared:

```
In [9]: pd.Int64Index([1, 2, 3]).equals(pd.UInt64Index([1, 2, 3]))                                                                                                                            
Out[9]: True
```

(in your example here, the values are not equal as a string is not equal to an integer)

**Review by jorisvandenbossche — COMMENTED** [review]
While you are at it, can you also add a "Parameters" section?

**Review by mroeschke — APPROVED** [review]

File: pandas/core/indexes/base.py [hunk 4]
```diff
@@ -4195,7 +4195,17 @@ def putmask(self, mask, value):
 
     def equals(self, other: Any) -> bool:
         """
-        Determine if two Index objects contain the same elements with the same order.
+        Determine if two Index object are equal.
+
+        The things that are being compared are:
+
+            * The elements inside the Index object.
+            * The order of the elements inside the Index object.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py:4203]:
```suggestion
        * The elements inside the Index object.
        * The order of the elements inside the Index object.
```

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**